Add IL2001-IL2060 trimmer warnings (#25842)
dotnet/docs: docs/core/deploying/trim-warnings/il2028.md

---
title: "IL2028: Known trimmer attribute does not have the required number of parameters"
description: "Learn about trim warning IL2028: TrimmerAttributeParametersArityMismatch"
ms.date: 08/25/2021
ms.topic: reference
author: mateoatr
ms.author: matorre
f1_keywords:
  - "IL2028"
---
# IL2028: Known trimmer attribute does not have the required number of parameters

## Cause

Trimmer found an instance of a known attribute that lacks the required constructor
parameters or has more than the accepted parameters. This can happen if a custom assembly

> defines a custom attribute whose fullname conflicts with the trimmer-known attributes,

Reviewer: ```suggestion
defines a custom attribute whose full name conflicts with the trimmer-known attributes,
```

> since the trimmer will recognizes custom attributes by doing a namespace and type name

Reviewer: ```suggestion
since the trimmer recognizes custom attributes by matching namespace and type name.
```

> match.

Reviewer: ```suggestion
```
